lib/api/core/models/repositories/roleRepository.js

```diff
@@ -29,30 +29,39 @@ module.exports = function (kuzzle) {
       deferred = q.defer(),
       keys = [],
       buffer = {},
-      result = [];
+      result = [],
+      internalIndexAllowedIds = {},
+      restrictions = {};
 
-    roleKeys.forEach((roleKey) => {
+    roleKeys.forEach(roleKey => {
```

Comment:
take a look at my proposal: https://github.com/kuzzleio/kuzzle/commit/3d22511d4632eb235ed1cac5936dcceba8821db4

If it's OK for you, you can merge directly my branch to yours ;)